pkg/controller/cloud/node_controller.go

```diff
@@ -147,6 +147,13 @@ func (cnc *CloudNodeController) updateNodeAddress(node *v1.Node, instances cloud
 		glog.V(5).Infof("This node %s is still tainted. Will not process.", node.Name)
 		return
 	}
+	//Node that isn't present according to the cloud provider shouldn't have its address updated
+	if exists, err := ensureNodeExistsByProviderIDOrExternalID(instances, node); err == nil {
+		if !exists {
+			glog.V(2).Infof("The node %s is no longer present according to the cloud provider, do not process.", node.Name)
+			return
+		}
+	}
```

Comment:
You should log the error.

Reply:
thanks, i fix like this. what your thinks about?
<pre>
// Node that isn't present according to the cloud provider shouldn't have its address updated
	exists, err := ensureNodeExistsByProviderIDOrExternalID(instances, node)
	if err != nil {
		glog.Errorf("%v", err)
	} else if !exists {
		glog.V(2).Infof("The node %s is no longer present according to the cloud provider, do not process.", node.Name)
		return
	}
</pre>

Reply:
```
// Node that isn't present according to the cloud provider shouldn't have its address updated
exists, err := ensureNodeExistsByProviderIDOrExternalID(instances, node)
if err != nil {
	glog.Errorf("%v", err)
	return
}

if !exists {
	glog.V(2).Infof("The node %s is no longer present according to the cloud provider, do not process.", node.Name)
	return
}
```
is what i'd do.